astroplan/scheduling.py

```diff
@@ -124,8 +126,170 @@ def components(self, val):
 
         self._components = val
         self.duration = duration
+        
+    @classmethod
+    @u.quantity_input(duration=u.second)
+    def from_duration(cls, duration):
+        # for testing how to put transitions between observations during
+        # scheduling without considering the complexities of duration
+        tb = TransitionBlock({None: 0*u.second})
+        tb.duration = duration
+        return tb
+
+
+class Schedule(object):
+    """
+    An object that represents a astronomical schedule
+    
+    Parameters:
+    -----------
+    start_time : `~astropy.time.Time`
+        The starting time of the schedule; the start of your 
+        observing window
+    end_time : `~astropy.time.Time`
+        The ending time of the schedule; the end of your
+        observing window
+    constraints : sequence of `Constraint`s
+        these are constraints that apply to the entire schedule
+    """
+    # as currently written, there should be no consecutive unoccupied slots
+    # this should change to allow for more flexibility (e.g. dark slots, grey slots)
+    
+    def __init__(self, start_time, end_time, constraints=None):
+        self.slots = [Slot(start_time, end_time)]
+        self.constraints = constraints
+        self.slew_duration = 4*u.min
+        # TODO: replace/overwrite slew_duration with Transitioner calls
 
+    def __repr__(self):
+        # once TransitionBlocks are defined, this should be scheduled_blocks
+        # TODO: make a better repr
+        for block in self.observing_blocks:
```

Comment:
I'm accessing a property  assigned below that using the same format as the `end_time` property of `TransitionBlock` defined on line 114.